cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1067,6 +1068,13 @@ def _add_head_node(self):
                     "instance_types_data_s3_key": f"{self.bucket.artifact_directory}/configs/instance-types-data.json",
                     "custom_node_package": self.config.custom_node_package or "",
                     "custom_awsbatchcli_package": self.config.custom_aws_batch_cli_package or "",
+                    "head_node_imds_secured": "true"
+                    if self.config.head_node.imds.secured
+                    or (
+                        self.config.head_node.imds.secured is None
+                        and self.config.scheduling.scheduler in SCHEDULERS_SUPPORTING_IMDS_SECURED
```

Comment:
We could in order to make the code cleaner, but I think the best place for this logic would rather be a function decorated with @property within the config object. I did not do it because `BaseClusterConfig` does not have information about the scheduler so we would have been forced to replicate it in every `[Scheduler]Config` and this is bad. I suppose this part will be covered by C&R. In the meantime I put the logic into a dedicated utility method just to make the code cleaner.